### DISCUSSION TITLE
Code Sample For Calling Graph API without Logging In

### DISCUSSION THREAD

**Discussion by leeenglestone** [microsoft/hack-together#59] [General]
Hi, Is there a C# code sample anywhere showing how to call the Graph API without making the user log in first?

Perhaps by creating some sort of API Key or Access Token in Azure and using that??

All the samples i've seen make the user log in..

Thanks,

-- Lee

**Comment by tarzzi** [comment]
You could look here for basics at authenticating with client secret credentials:
https://learn.microsoft.com/en-us/graph/auth-v2-service

And here for some samples how to implement this in C#
https://learn.microsoft.com/fi-fi/graph/sdks/choose-authentication-providers?tabs=CS#client-credentials-provider

Hope these help! 😊

**Reply by leeenglestone** [reply]
Thanks - I'll have a look and give it a go

**Reply by aycabas** [reply]
Thanks @tarzzi for helping on this! @leeenglestone if you need further assistance, let us know :)

**Reply by leeenglestone** [reply]
When I log into the tenant that I created for the hackathon by following the instructions, it says I am a "Restricted Tenant" and when I go to Azure Active Director to create an app it says "You do not have access".

Please advise, as currently I can't create an app to use to call the graph api. Thanks,

![image](https://user-images.githubusercontent.com/2360725/223405248-5156dc27-7c39-4db1-8d6b-9b5572afe3e1.png)

@tarzzi @aycabas

**Reply by leeenglestone** [reply]
I also see this in the portal which doesn't look quite right..

![image](https://user-images.githubusercontent.com/2360725/223405902-fb322bc5-f1ba-4336-8b95-58abd6238b39.png)

**Reply by leeenglestone** [reply]
Ahh.. I've got a bit further by switching directories in this tenant from the default one, to one called MSFT.

**Reply by leeenglestone** [reply]
For anyone else that is going down this route and you are getting OData errors like described here
[https://stackoverflow.com/questions/74775906/some-methods-in-graphserviceclient-not-available-microsoft-graph](https://stackoverflow.com/questions/74775906/some-methods-in-graphserviceclient-not-available-microsoft-graph)

Note, you cannot call "Me" to get the Users details but you can get other details if you give the right permissions
Also specifically catching and displaying/interrogating the OData error information really helps
![image](https://user-images.githubusercontent.com/2360725/223415241-68109896-fb61-45dc-b409-2fba212f3804.png)
